File: DownloadFiles/TGNet/Chapter11/VB/Invoice.vb
Public Class Invoice
    Private _InvoiceNumber As Int32
    Private _InvoiceDate As Date
    Private _CustomerID As Int32
    Private _TotalSales As Decimal
    Private Shared LastInvoice As Int32

    Public Sub New(ByVal CustID As Int32)
        _CustomerID = CustID
        _InvoiceNumber = NewInvoiceNumber()
    End Sub

    Public ReadOnly Property InvoiceNumber() As Int32
        Get
            Return _InvoiceNumber
        End Get
    End Property
    Public Property InvoiceDate() As Date
        Get
            Return _InvoiceDate
        End Get
        Set(ByVal value As Date)
            _InvoiceDate = value
        End Set
    End Property
    Public Property CustomerID() As Int32
        Get
            Return _CustomerID
        End Get
        Set(ByVal value As Int32)
            _CustomerID = value
        End Set
    End Property

    Public Function CalculateInvoiceTotal(ByVal Sale As Decimal) As Decimal      
        
    End Function

    Public Function NewInvoiceNumber() As Int32
        'open a database and retrieve highest invoice number
        LastInvoice += 1
        Return LastInvoice
    End Function
End Class

File: DownloadFiles/TGNet/Chapter11/C#/Invoice.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace InvoiceLib
{
    public class Invoice
    {
        private Int32 _InvoiceNumber;
        private System.DateTime _InvoiceDate;
        private Int32 _CustomerID;
        private static Int32 LastInvoice;

        public Invoice(Int32 CustID)
        {
            _CustomerID = CustID;
            _InvoiceNumber = NewInvoiceNumber();
        }

        public Int32 InvoiceNumber
        {
            get
            {
                return _InvoiceNumber;
            }
        }

        public System.DateTime InvoiceDate
        {
            get
            {
                return _InvoiceDate;
            }
            set
            {
                _InvoiceDate = value;
            }
        }

        public Int32 CustomerID
        {
            get
            {
                return _CustomerID;
            }
            set
            {
                _CustomerID = value;
            }
        }

        public decimal CalculateInvoiceTotal()
        {
            
        }

        public Int32 NewInvoiceNumber()
        {
            LastInvoice += 1;
            return LastInvoice;
        }
    }
}
